lib/ansible/parsing/dataloader.py

```diff
@@ -233,35 +236,33 @@ def path_dwim(self, given):
         given = unquote(given)
         given = to_text(given, errors='surrogate_or_strict')
 
-        if given.startswith(u"/"):
-            return os.path.abspath(given)
-        elif given.startswith(u"~"):
-            return os.path.abspath(os.path.expanduser(given))
+        if RE_ABS_PATH.search(given):
```

Comment:
it depends on the data.  For regex, the longer the string the quicker performance drops off.  here's some testing:
``` python
>>> RE = re.compile('(?:a|b)')
>>> def match(test):
...  if test.endswith('a') or test.endswith('b'): return test
... 
>>> def match2(test):
...  if RE.search(test): return test
>>> timeit.repeat('match("1" * 1)', 'from __main__ import match')
[0.321303129196167, 0.3136630058288574, 0.31526708602905273]
>>> timeit.repeat('match2("1" * 1)', 'from __main__ import match2')
[0.32927894592285156, 0.3168821334838867, 0.31973695755004883]
>>> timeit.repeat('match("1" * 10)', 'from __main__ import match')
[0.31588292121887207, 0.3094758987426758, 0.31125402450561523]
>>> timeit.repeat('match2("1" * 10)', 'from __main__ import match2')
[0.40798497200012207, 0.4003629684448242, 0.3933870792388916]
>>> timeit.repeat('match("1" * 100)', 'from __main__ import match')
[0.37717509269714355, 0.3657679557800293, 0.3643019199371338]
>>> timeit.repeat('match2("1" * 100)', 'from __main__ import match2')
[1.2596900463104248, 1.2499568462371826, 1.239171028137207]
```

Unless your string sizes are very small, the rule of thumb should probably be that regex is never faster than any number of substring matches.